Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -48,6 +48,15 @@ def setup_build_environment(formula: nil, cc: nil, build_bottle: false, bottle_a
     @build_bottle = build_bottle
     @bottle_arch = bottle_arch
     reset
+
+    return unless formula
+
+    source_date_epoch = if (tap = formula.tap)
+      Utils.popen_read("git", "-C", tap.path.to_s, "log", "-1", "--format=%ct", formula.path.to_s).to_i
+    else
+      File.mtime(formula.path).to_i
+    end
+    self["SOURCE_DATE_EPOCH"] = source_date_epoch.to_s if source_date_epoch.positive?
```

Comment:
Would be good to use `source_modified_time` here and ensure consistency/maybe share some logic with https://github.com/Homebrew/brew/blob/bc9b98aa0ba777889218ac51f67ccc76b0be6a28/Library/Homebrew/dev-cmd/bottle.rb#L390-L410

Reply:
`Formula#source_modified_time` appears to be based on the mtime of the downloaded file.

I suppose it depends on whether that's what we want or if it's better to have the modified time of the formula file.

Reply:
> `Formula#source_modified_time` appears to be based on the mtime of the downloaded file.

Yes, this is what we want, I think.



> I suppose it depends on whether that's what we want or if it's better to have the modified time of the formula file.

I think that's a better fit. If the e.g. `desc` changes on the formula I don't see why we'd want to generate a binary package with a different checksum.